clustering/ejb/WFLY-7628_Distributed_EJB_Timers.adoc

```diff
@@ -0,0 +1,168 @@
+* Add ability to enable distributed timer management (in lieu of a timer store) within ejb3 subsystem for management of persistent timers.
+* Existing timer service resource operations must continue to work when executed against the distributable `TimerService`.
+
+=== Nice-to-Have Requirements
+
+* Add default and named transient timer management capabilities to distributable-ejb subsystem.
+* Add an attribute to the timer service resource within the ejb3 subsystem to determine the server-wide default provider for transient timer management.
+* Add ability to switch between distributed timer management or existing in-memory implementation for management of transient timers
+** This has the advantage of supporting passivation of timer instances to support applications with a sufficiently large number of transient timers in excess of some configured maximum.
+* Add support to timer service deployment descriptor namespace for overriding persistent/transient timer management per EJB.
+
+=== Non-Requirements
+
+* This feature will not change the default timer service behavior or deprecate any existing timer service mechanisms.
+
+== Backwards Compatibility
+
+* Servers started using the default ejb3 subsystem configuration will continue to behave as normal, using the existing in-memory `TimerService` implementation for both transient and persistent timers.
+
+=== Default Configuration
+
+The following caches will be added to the default `ejb` cache-container configuration with the Infinispan subsystem:
+
+For the HA profiles:
+
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent:add()
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent/component=locking:add(isolation=PESSIMISTIC)
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent/component=transaction:add(mode=BATCH)
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent/store=file:add(purge=false)
+
+And for non-HA profiles:
+
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent:add()
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent:component=locking:add(isolation=PESSIMISTIC)
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent:component=transaction:add(mode=BATCH)
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent/store=file:add(purge=false, passivation=false)
+
+The following timer management provider will be added to the default distributable-ejb subsystem configuration:
+
+	/subsystem=distributable-ejb/infinispan-timer-management=persistent:add(cache-container=ejb, cache=persistent)
+
+To enable use of the distributable timer service, a user will be required to modify the default timer-service configuration of the ejb3 subsystem as follows:
+
+	/subsystem=ejb3/service=timer-service:undefine-attribute(name=default-data-store)
+	/subsystem=ejb3/service=timer-service:write-attribute(name=default-persistent-timer-management=persistent)
```

Comment:
This line looks broken, maybe it should read as follows?

```
/subsystem=ejb3/service=timer-service:write-attribute(name=default-persistent-timer-management, value=persistent)
```

Reply:
Yeah - that's a typo.